### PR TITLE
#159965313 Email notifications

### PR DESCRIPTION
**Pull request by Bileonaire**
#### What does this PR do?
- Enable notifications

#### Description of task to be completed?
- Enable users to receive email notifications when users they follow publish new articles.

#### How should this be manually tested?
$ git clone https://github.com/andela/ah-magnificent6.git
$ git checkout ft-receive-notification- 159965313
$ cd ah-magnificent6
$ cp .env-sample .env [and set the environment variables]
$ python3 -m venv venv
$ source venv/bin/activate
(venv) $ pip install -r requirements.txt
(venv) $ python manage.py migrate
(venv) $ python manage.py runserver

Create two users and activate them, make one follow the other. 
Create an article with the one being followed.

Using [Postman](https://www.getpostman.com/) navigate to 
User registration: `POST {your-server-root}:8000/api/user/signup`
Create two users and activate them
```
{
    "email": "author@user.user",
    "username": "author",
    "password": "good123456"
}
```

User Login: `POST {your-server-root}:8000/api/user/login`
```
{
    "email": "author@user.user",
    "password": "good123456"
}
```
Use the token from the response in the login as the authentication header while creating an article.

Create article : POST {your-server-root}:8000/api/articles/
```
{
"title": "My Journey to Andela",
 "description": "This article is about how I joined Andela",
"body": "This is my story to Andela"
}
```
run command
`python manage.py runcrons`

or 
`python manage.py runcrons --force`
to run cron regardless of the set time in the cron job.

The email notification will be sent by the cron-job

screen shot 2018-09-24 at 16 45 59

### CODE REVIEW

File: authors/apps/core/cron.py [hunk 1]
```diff
@@ -1,32 +1,39 @@
...
-                        notification.save()
+                    if user not in notification.read.all():
+                        if user.profile.email_notification_enabled == True:
+                            emails.append(user.email)
+                content = {'notification': notification}
+                message = render_to_string("notification.html", content)
+                mail = EmailMessage(
+                    subject=subject,
+                    body=message,
+                    to=emails,
```
Comment (pngg) [inline authors/apps/core/cron.py]: Could we rename `emails` to `recipients` to reduce ambiguity?

Comment (Bileonaire) [reply authors/apps/core/cron.py]: Very true. Done. @pngg

**Comment by mikenthiwa** [comment]
@Bileonaire Looks good

File: authors/apps/core/cron.py [hunk 2]
```diff
@@ -1,32 +1,39 @@
...
     schedule = Schedule(run_every_mins=RUN_EVERY_MINS)
     code = 'authors.apps.core.cron.MyCronJob'  # a unique code
 
     def do(self):
-        """send emails to all the persons to be notified"""
+        """Send emails to all the persons to be notified."""
         subject = 'Authors Haven Notification'
 
-        emails = []
+        receipients = []
```
Comment (parseen254) [inline authors/apps/core/cron.py]: @Bileonaire grammar police will get you. recipient not receipients

**Review by parseen254 — CHANGES_REQUESTED** [review]

**Review by parseen254 — APPROVED** [review]

**Pull request merged**